SPDX-License-Identifier: GPL-2.0-or-later

Use mkstemp(3) for the print temporary file instead of tmpnam(3).

Index: source/file.c
--- source/file.c.orig
+++ source/file.c
@@ -1373,7 +1373,12 @@
 */
 void PrintString(const char *string, int length, Widget parent, const char *jobName)
 {
+#ifdef HAVE_MKSTEMP
+    static char template[] = "/var/tmp/nedit-XXXXXXXXXX";
+    char tmpFileName[sizeof(template)];
+#else
     char tmpFileName[L_tmpnam];    /* L_tmpnam defined in stdio.h */
+#endif
     FILE *fp;
     int fd;
 
@@ -1384,6 +1389,10 @@
 	    1. Create a filename
 	    2. Open the file with the O_CREAT|O_EXCL flags
 	So all an attacker can do is a DoS on the print function. */
+#ifdef HAVE_MKSTEMP
+    strcpy(tmpFileName, template);
+    if ((fd = mkstemp(tmpFileName)) < 0 || (fp = fdopen(fd, "w")) == NULL)
+#else
     tmpnam(tmpFileName);
 
     /* open the temporary file */
@@ -1392,6 +1401,7 @@
 #else
     if ((fd = open(tmpFileName, O_CREAT|O_EXCL|O_WRONLY, S_IRUSR | S_IWUSR)) < 0 || (fp = fdopen(fd, "w")) == NULL)
 #endif /* VMS */
+#endif /* HAVE_MKSTEMP */
     {
         DialogF(DF_WARN, parent, 1, "Error while Printing",
                 "Unable to write file for printing:\n%s", "OK",
